src/nvim/strings.c

```diff
@@ -496,6 +496,20 @@ char *vim_strchr(const char *const string, const int c)
   }
 }
 
+// Sized version of strchr that can handle embedded NULs.
+// Adjusts n to the new size.
+char *strnchr(register const char *p, size_t *n, int c)
```

Comment:
no need for `register` annotation on variables.

Reply:
How come? I copied this from GCC's `strchr` implementation, so if they think it's necessary, wouldn't it also be necessary here too?

Reply:
yes, and when we copy in code into our codebase we adjust it to follow _our_ conventions (which is based on what compilers _we_ support, etc). We don't use `register` annotations in general (because we trust the optimizer with that), and I don't think this function needs to be marked that it came originally from a codebase which uses them.

Reply:
I compiled this with `gcc` and `clang`. `gcc` with `register` produces the least amount of code.

<details>
<summary>`gcc` (no `register`) - 28 instructions</summary>
```
d10083ff     	sub	sp, sp, #0x20
f9000fe0     	str	x0, [sp, #0x18]
f9000be1     	str	x1, [sp, #0x10]
b9000fe2     	str	w2, [sp, #0xc]
14000011     	b	0x54 <ltmp0+0x54>
f9400fe0     	ldr	x0, [sp, #0x18]
39c00000     	ldrsb	w0, [x0]
2a0003e1     	mov	w1, w0
b9400fe0     	ldr	w0, [sp, #0xc]
6b01001f     	cmp	w0, w1
54000061     	b.ne	0x34 <ltmp0+0x34>
f9400fe0     	ldr	x0, [sp, #0x18]
1400000e     	b	0x68 <ltmp0+0x68>
f9400fe0     	ldr	x0, [sp, #0x18]
91000400     	add	x0, x0, #0x1
f9000fe0     	str	x0, [sp, #0x18]
f9400be0     	ldr	x0, [sp, #0x10]
b9400000     	ldr	w0, [x0]
51000401     	sub	w1, w0, #0x1
f9400be0     	ldr	x0, [sp, #0x10]
b9000001     	str	w1, [x0]
f9400be0     	ldr	x0, [sp, #0x10]
b9400000     	ldr	w0, [x0]
7100001f     	cmp	w0, #0x0
54fffdac     	b.gt	0x14 <ltmp0+0x14>
d2800000     	mov	x0, #0x0                ; =0
910083ff     	add	sp, sp, #0x20
d65f03c0     	ret
```
</details>

<details>
<summary>`gcc` (with `register`) - 23 instructions</summary>
```
d10043ff     	sub	sp, sp, #0x10
f90007e1     	str	x1, [sp, #0x8]
b90007e2     	str	w2, [sp, #0x4]
1400000d     	b	0x40 <ltmp0+0x40>
39c00001     	ldrsb	w1, [x0]
2a0103e2     	mov	w2, w1
b94007e1     	ldr	w1, [sp, #0x4]
6b02003f     	cmp	w1, w2
54000041     	b.ne	0x28 <ltmp0+0x28>
1400000c     	b	0x54 <ltmp0+0x54>
91000400     	add	x0, x0, #0x1
f94007e1     	ldr	x1, [sp, #0x8]
b9400021     	ldr	w1, [x1]
51000422     	sub	w2, w1, #0x1
f94007e1     	ldr	x1, [sp, #0x8]
b9000022     	str	w2, [x1]
f94007e1     	ldr	x1, [sp, #0x8]
b9400021     	ldr	w1, [x1]
7100003f     	cmp	w1, #0x0
54fffe2c     	b.gt	0x10 <ltmp0+0x10>
d2800000     	mov	x0, #0x0                ; =0
910043ff     	add	sp, sp, #0x10
d65f03c0     	ret
```
</details>

<details>
<summary>`clang` (with and without `register`) - 34 instructions</summary>
```
d10083ff     	sub	sp, sp, #0x20
f9000be0     	str	x0, [sp, #0x10]
f90007e1     	str	x1, [sp, #0x8]
b90007e2     	str	w2, [sp, #0x4]
14000001     	b	0x14 <ltmp0+0x14>
f94007e8     	ldr	x8, [sp, #0x8]
b9400108     	ldr	w8, [x8]
71000108     	subs	w8, w8, #0x0
1a9fc7e8     	cset	w8, le
37000288     	tbnz	w8, #0x0, 0x74 <ltmp0+0x74>
14000001     	b	0x2c <ltmp0+0x2c>
f9400be8     	ldr	x8, [sp, #0x10]
39c00108     	ldrsb	w8, [x8]
b94007e9     	ldr	w9, [sp, #0x4]
6b090108     	subs	w8, w8, w9
1a9f07e8     	cset	w8, ne
370000a8     	tbnz	w8, #0x0, 0x54 <ltmp0+0x54>
14000001     	b	0x48 <ltmp0+0x48>
f9400be8     	ldr	x8, [sp, #0x10]
f9000fe8     	str	x8, [sp, #0x18]
1400000b     	b	0x7c <ltmp0+0x7c>
f9400be8     	ldr	x8, [sp, #0x10]
91000508     	add	x8, x8, #0x1
f9000be8     	str	x8, [sp, #0x10]
f94007e9     	ldr	x9, [sp, #0x8]
b9400128     	ldr	w8, [x9]
71000508     	subs	w8, w8, #0x1
b9000128     	str	w8, [x9]
17ffffe9     	b	0x14 <ltmp0+0x14>
f9000fff     	str	xzr, [sp, #0x18]
14000001     	b	0x7c <ltmp0+0x7c>
f9400fe0     	ldr	x0, [sp, #0x18]
910083ff     	add	sp, sp, #0x20
d65f03c0     	ret
```
</details>